tests/atoms/gaia_data_layer.js

```diff
@@ -45,16 +45,17 @@ var GaiaDataLayer = {
 
   disableBluetooth: function() {
     var bluetooth = window.navigator.mozBluetooth;
-    if (bluetooth.enabled) {
+    if (bluetooth.defaultAdapter.state == 'enabled') {
       console.log('trying to disable bluetooth');
+      bluetooth.defaultAdapter.disable();
       this.setSetting('bluetooth.enabled', false, false);
       waitFor(
```

Comment:
If the adapter is already in the `disabling` at the time it reaches `if (bluetooth.defaultAdapter.state == 'enabled')`, the wait will never happen